### DISCUSSION TITLE
How to impement a fully adjacent layer in GCN?

### DISCUSSION THREAD

**Discussion by codeastra2** [pyg-team/pytorch_geometric#6887] [General]
I would like to implement a [fully adjacent layer](https://arxiv.org/abs/2006.05205) in GNN models. i currently have an architecture like so:

```
        self.conv1 = GCNConv(num_node_features, 16)
        self.conv2 = GCNConv(16, 16)
        self.conv3 = GCNConv(16, 16)
        self.conv4 = GCNConv(16, 16)
```

How can I make conv4 have a fully adjacent network?

**Comment by rusty1s** [comment]
You mean a fully-connected graph? That is determined by the input, i.e., `edge_index`. You can generate a fully connected graph via
```python
edge_index = torch.ones(num_nodes, num_nodes).nonzero().t()
```
but keep in mind that this may be slow to run since we are basically using an operator designed for sparse inputs on a fully-connected graph.
